src/js/filter.ts

```diff
@@ -1,19 +1,56 @@
 /* global document, window */
 
+import type Choices from "choices.js";
 import type { CircleMarker, FeatureGroup } from "leaflet";
 import type { CityId, CityEntry } from "./types";
 
+/**
+ * Return true if the city should be rendered on the map.
+ *
+ * Note that search takes priority. If certain cities are selected via
+ * search, they should be shown regardless of the filters.
+ */
+const shouldBeSelected = (
+  cityState: CityId,
+  entry: CityEntry,
+  searchElement: Choices
+): boolean => {
+  const searchChosen = new Set(searchElement.getValue(true) as string[]);
+  if (searchChosen.size > 0) {
+    return searchChosen.has(cityState);
+  }
+
+  // Else, search is not used and the filters should apply.
+  const getSelected = (cls: string): Set<string> =>
+    new Set(
+      Array.from(document.querySelectorAll(cls)).map(
+        (option: HTMLInputElement) => option.value
+      )
+    );
+
+  const scopeSelected = getSelected(".filter--scope :checked");
+  const isScope = entry["report_magnitude"]
+    .split(",")
+    .some((scope) => scopeSelected.has(scope));
+
+  return isScope;
```

Comment:
Is the plan to add additional filters through here?

Reply:
Exactly